### DISCUSSION TITLE
update multiple form fields in child component

### DISCUSSION THREAD

**Discussion by piyushmani** [react-hook-form/react-hook-form#7713] [Q&A]
I have a parent component where I need to update around 30 fields and I have create a child component to manage 4 field out of them using some visual representation .

```
<Controller
          control={control}
          render={({ ref, field }) => (
          <PersonHeightSelect {...field} ref={null} />)}
   /> 
```

Can you please help how can I update the one or all four property in child component. 
I tried calling like this onChange({'prop':val}) in child component but it does not work

**Comment by JamesWong1999** [comment]
use [setValue](https://react-hook-form.com/api/useform/setvalue) API to update the other fields?

```tsx
onChange={(e) => {
  setValue('otherField1', 'data1');
  setValue('otherField2', 'data2');
  setValue('otherField'3, 'data3');
  setValue('otherField'4, 'data4');
  field.onChange(e.target.value)
}}
```

**Reply by piyushmani** [reply]
what will happen at last line ?
field.onChange(e.target.value)
will it update all 4 attribute ?

**Reply by JamesWong1999** [reply]
The last line is to update your component value, the other 4 `setValue` will update those values outside of the `Controller` component.

![Screen Shot 2022-02-02 at 3 36 22 pm](https://user-images.githubusercontent.com/44851854/152093823-652b3c54-4860-42c4-b658-167d2f19972c.png)

reference to the doc: https://react-hook-form.com/api/usecontroller/controller

**Comment by Moshyfawn** [comment]
#2363
